gradle/libs.versions.toml

```diff
@@ -0,0 +1,432 @@
+gulp-cli = "2.3.0"
+hamcrest = "2.2"
+hk2 = "3.0.5"
+hsqldb = "2.7.2"
+ibm-icu = "74.2"
+immutables-valueannotations = "2.10.1"
+j256-simplemagic = "1.17"
+jakarta-annotation = "2.1.1"
+jakarta-inject = "2.0.1"
+jakarta-ws = "3.1.0"
+javacc = "7.0.12"
+# @keep for version alignment
+jaxb = "2.3.8"
+jayway-jsonpath = "2.9.0"
+jctools = "4.0.5"
+jersey = "3.1.5"
+# TODO Sync with jersey versions
+jersey-containers = "2.39.1"
+# @keep for version alignment
+joda-time = "2.8.1"
+junit = "4.13.2"
+# @keep Link checker version used in ref-guide
+link-checker = "1.4.2"
+littlerobots-versioncatalogupdate = "0.8.4"
+lmax-disruptor = "3.4.4"
+ltgt-errorprone = "3.1.0"
+# @keep This is the minimum required Java version.
+minJava = "11"
+mockito = "5.12.0"
+morethan-jmhreport = "0.9.0"
+navsecurity = "0.5.10"
+netty = "4.1.112.Final"
+# @keep for version alignment
+netty-tcnative = "2.0.65.Final"
+nimbusds-josejwt = "9.30.2"
+nodegradle-node = "7.0.1"
+openapi = "7.6.0"
+openjdk-jmh = "1.37"
+opentelemetry = "1.40.0"
+osgi-annotation = "8.1.0"
+# @keep for version alignment
+ow2-asm = "9.3"
+owasp-dependencycheck = "9.0.8"
+# @keep for version alignment
+perfmark = "0.27.0"
+prometheus-metrics = "1.1.0"
+prometheus-simpleclient = "0.16.0"
+quicktheories = "0.26"
+semver4j = "5.3.0"
+slf4j = "2.0.13"
+spatial4j = "0.8"
+spotbugs = "4.8.0"
+squareup-okhttp3-mockwebserver = "4.11.0"
+squareup-okhttp3-okhttp = "4.12.0"
+stephenc-jcip = "1.0-1"
+swagger3 = "2.2.22"
+tdunning-tdigest = "3.3"
+thetaphi-forbiddenapis = "3.7"
+thisptr-jacksonjq = "0.0.13"
+threeten-bp = "1.6.8"
+undercouch-download = "5.5.0"
+xerces = "2.12.2"
+xerial-snappy = "1.1.10.5"
+
+[plugins]
+benmanes-versions = { id = "com.github.ben-manes.versions", version.ref = "benmanes-versions" }
+carrotsearch-dependencychecks = { id = "com.carrotsearch.gradle.dependencychecks", version.ref = "carrotsearch-dependencychecks" }
+cutterslade-analyze = { id = "ca.cutterslade.analyze", version.ref = "cutterslade-analyze" }
+diffplug-spotless = { id = "com.diffplug.spotless", version.ref = "diffplug-spotless" }
+littlerobots-versioncatalogupdate = { id = "nl.littlerobots.version-catalog-update", version.ref = "littlerobots-versioncatalogupdate" }
+ltgt-errorprone = { id = "net.ltgt.errorprone", version.ref = "ltgt-errorprone" }
+morethan-jmhreport = { id = "io.morethan.jmhreport", version.ref = "morethan-jmhreport" }
+nodegradle-node = { id = "com.github.node-gradle.node", version.ref = "nodegradle-node" }
+openapi-generator = { id = "org.openapi.generator", version.ref = "openapi" }
+owasp-dependencycheck = { id = "org.owasp.dependencycheck", version.ref = "owasp-dependencycheck" }
+swagger3-core = { id = "io.swagger.core.v3.swagger-gradle-plugin", version.ref = "swagger3" }
+thetaphi-forbiddenapis = { id = "de.thetaphi.forbiddenapis", version.ref = "thetaphi-forbiddenapis" }
+udnercouch-download = { id = "de.undercouch.download", version.ref = "undercouch-download" }
+
+[libraries]
```

Comment:
ah well, we've got to follow what Gradle wants us to do now but IMO coming up with a unique library string for each library is annoying.  Maintaining this is annoying.

Reply:
You have to come up with your own name. Intellij will help you with suggestion prompts, if you're using it but otherwise you're down to using those custom names.

Reply:
I agree with this. Lack of namespacing will get us into trouble. Would it be better if we agree on some simple convention to use the original coordinate with dots replaced by space? E.g.

```toml
[versions]
org-apache-commons_commons-compress = "1.26.1"
org-apache-lucene = "9.12.0"

[libraries]
org-apache-commons_commons-compress = { module = "org.apache.commons:commons-compress", version.ref = "org-apache-commons_commons-compress" }
org-apache-lucene_lucene-foo = { module = "org.apache.lucene:lucene-foo", version.ref = "org-apache-lucene" }
org-apache-lucene_lucene-bar = { module = "org.apache.lucene:lucene-bar", version.ref = "org-apache-lucene" }

```

Reply:
I think before we run into namespace limits we would have other issues first. 😅 However, even if a collection of libraries use the same group and eventually similar artifact namings, we always can add more words in the alias, like prefixes and suffixes. This is already the case for some libraries.

Using the fully qualified name would require renaming the alias and all places where it is used if a library changes group or artifact name (like when splitting into multiple modules). So I would still prefer a simplified alias, rather than the fully qualified name.

Reply:
You can do that, although there is a meaning to prefixes like this in gradle catalogs: they form artifact groups [1]. When I started using them, I followed a convention similar to yours... but over time I resigned from doing that. It wasn't really that helpful (or intuitive, with those dashes everywhere). 

[1] https://docs.gradle.org/current/userguide/platforms.html#sub:mapping-aliases-to-accessors